lib/internal/bootstrap_node.js

```diff
@@ -361,20 +361,19 @@
         }
 
       } else {
-        // If we handled an error, then make sure any ticks get processed
-        NativeModule.require('timers').setImmediate(process._tickCallback);
-
         // Emit the after() hooks now that the exception has been handled.
         if (async_hook_fields[kAfter] > 0) {
           do {
             NativeModule.require('async_hooks').emitAfter(
               async_uid_fields[kCurrentAsyncId]);
-          // popAsyncIds() returns true if there are more ids on the stack.
-          } while (popAsyncIds(async_uid_fields[kCurrentAsyncId]));
+          } while (asyncIdStackSize() > 0);
```

Comment:
>>who actually pops the ids?
>
>emitAfter() pops the id. that's why it was skipping every other id in the stack.

Ohh `emitAfter` is `emitAfterScript` not `emitAfterNative`

>> cache emitAfter?
>
>I'm doing it for the same reason as the require('timers') call below. Because if the script fails early in the startup process then the module might not have been loaded yet. And require() already caches the script into an object, doesn't it?

Ack, but if you do it just before the loop it (1) looks cleaner [to me] (2) still lazy enough.